.github/workflows/push.yml

```diff
@@ -0,0 +1,49 @@
+name: Single Build & Test (Push)
+
+# workflow is for a branch push only. do not run on push to master.
+# push to master (which is a pull requet merge) has a more elaborate workflow to run
+# github repo is configured with branch protection on master.
+on: 
+  push:
+    branches-ignore:
+      - 'master'
+
+  workflow_dispatch:
+    inputs: 
+      node:
+        required: false
+        description: 'A node image (e.g: node:15.14.0) or docker dir (e.g. ./.github/docker-node/16/centos7)'
+        default: 'node' # as of June 2021 this is node:16-buster
+
+jobs:
+  build-single-test:
```

Comment:
This specific workflow, can in theory, run on the *runner*. It will require to add https://github.com/actions/setup-node. I *strongly* prefer to keep ALL workflows with the same container generation method. Also - the ability to manually select a specific image has already proven useful...